src/frontend/src/expr/literal.rs

```diff
@@ -55,6 +58,24 @@ impl Literal {
     pub fn get_data(&self) -> &Datum {
         &self.data
     }
+
+    pub fn is_null(&self) -> bool {
+        self.data.is_none()
+    }
+
+    /// This is a temporary workaround.
+    ///
+    /// TODO: evaluate expression in frontend.
+    /// Tracking issue <https://github.com/singularity-data/risingwave/issues/3479>
+    pub fn eval_as(&self, return_type: DataType) -> ScalarImpl {
+        assert!(self.data.is_some());
+        let lit = LiteralExpression::try_from(&self.to_expr_proto()).unwrap();
+        let mut const_expr = lit.boxed();
+        if const_expr.return_type() != return_type {
+            const_expr = new_unary_expr(ExprType::Cast, return_type, const_expr).unwrap();
+        }
+        const_expr.eval_row(Row::empty()).unwrap().unwrap()
+    }
```

Comment:
lgtm. Just to provide an alternative:

As suggested by @neverchanje offline, we could approach #3479 in a similar way:
```
impl ExprImpl {
    pub fn eval_row(&self, input: &Row) -> Result<Datum> {
        let be_expr = build_from_prost(&self.to_expr_proto())?;
        be_expr.eval_row(input)
    }
}
```
This interface is unlikely to change, and we can replace its implementation to avoid proto ser/de later.

At the caller side, we can:
```
lit.cast_implicit(return_type)?.eval_row(Row::empty())
```
There would be no `is_null` or `eval_as` on `Literal`.


Supporting const_expr beyond literals can also build on top of this.

---

The use of `cast_implicit` also reminds us of the following case:
```
select * from t where v1 < 60000 -- v1 is actually smallint
```
This should evaluate to true. But casting 60000 to smallint would result in an error. `cast_implicit` would report the error in all cases even when it is `v1 < 2` forcing us to handle this case, while `eval_as` only reports the error when it actually exceeds `32767`.

Reply:
I was also considering this. Do it later